assets/src/blocks/SocialMedia/SocialMediaEditorScript.js

```diff
@@ -0,0 +1,250 @@
+import { useEffect } from '@wordpress/element';
+import { InspectorControls } from '@wordpress/block-editor';
+import {
+  RadioControl,
+  SelectControl,
+  PanelBody,
+} from '@wordpress/components';
+import { SocialMediaEmbed } from './SocialMediaEmbed';
+import { URLInput } from '../../components/URLInput/URLInput';
+import { HTMLSidebarHelp } from '../../components/HTMLSidebarHelp/HTMLSidebarHelp';
+import {
+  OEMBED_EMBED_TYPE,
+  FACEBOOK_EMBED_TYPE,
+  FACEBOOK_PAGE_TAB_TIMELINE,
+  FACEBOOK_PAGE_TAB_EVENTS,
+  FACEBOOK_PAGE_TAB_MESSAGES,
+  ALLOWED_OEMBED_PROVIDERS,
+} from './SocialMediaConstants.js';
+
+const { RichText } = wp.blockEditor;
+const { __ } = wp.i18n;
+const { apiFetch } = wp;
+const { addQueryArgs } = wp.url;
+
+const loadScriptAsync = uri => {
+  return new Promise((resolve, reject) => {
+    let tag = document.createElement('script');
+    tag.src = uri;
+    tag.async = true;
+    tag.onload = () => {
+      resolve();
+    };
+    const body = document.getElementsByTagName('body')[0];
+    body.appendChild(tag);
+  });
+};
+
+const initializeTwitterEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.twttr) {
+      window.twttr.widgets.load();
+    }
+  }, 2000);
+}
+
+const initializeInstagramEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.instgrm) {
+      window.instgrm.Embeds.process();
+    }
+  }, 3000);
+}
+
+const initializeFacebookEmbeds = () => {
+  setTimeout(() => {
+    if ('undefined' !== window.FB) {
+      window.FB.XFBML.parse();
+    }
+  }, 3000);
+}
+
+const PROVIDER_SCRIPT_DATA = {
+  twitter: {
+    script: 'https://platform.twitter.com/widgets.js',
+    initFunction: initializeTwitterEmbeds,
+  },
+  facebook: {
+    script: 'https://connect.facebook.net/en_US/sdk.js#xfbml=1&version=v9.0',
+    initFunction: initializeFacebookEmbeds,
+  },
+  instagram: {
+    script: 'https://www.instagram.com/embed.js',
+    initFunction: initializeInstagramEmbeds,
+  },
+};
+
+export const SocialMediaEditor = ({
+  attributes,
+  isSelected,
+  setAttributes,
+}) => {
+  const {
+    social_media_url,
+    facebook_page_tab,
+    description,
+    title,
+    embed_type,
+    alignment_class,
+    embed_code,
+    className,
+  } = attributes;
+
+  const toAttribute = attributeName => value => setAttributes({
+    [attributeName]: value
+  });
+
+  /**
+   * Check if social media corresponding embeds script is loaded and initiliaze it.
+   * Can be used for Facebook, Twitter and Instagram depending on the parameter.
+   */
+  const checkProviderScript = async provider => {
+    const providerData = PROVIDER_SCRIPT_DATA[provider];
+    const script = document.querySelector(`body > script[src="${providerData.script}"]`);
+    if (script === null) {
```

Comment:
`providerData.initFunction` is called in both cases, so it can be after the if-statement once.